source/common/grpc/common.cc

```diff
@@ -112,6 +112,65 @@ Status::GrpcStatus Common::httpToGrpcStatus(uint64_t http_response_status) {
   }
 }
 
+uint64_t Common::grpcToHttpStatus(Status::GrpcStatus grpc_status) {
```

Comment:
This actually comes straight from an internal doc on the canonical gRPC -> HTTP mapping. In terms of public documentation, there are HTTP -> gRPC mappings at https://github.com/grpc/grpc/blob/master/doc/http-grpc-status-mapping.md, but not the other way.

Digging around, it looks like this mapping as reflected in a table at https://cloud.google.com/apis/design/errors#handling_errors. 

I will add a comment pointing at that. In terms of test, what are you after? Since this function is basically a table, it seems the unit test would largely resemble the implementation.

Reply:
Comment would be great, thanks. A test would probably look like this: https://github.com/envoyproxy/envoy/blob/master/test/common/http/codes_test.cc#L94. I agree that test is dubiously useful, but it does add some value. Up to you.

Reply:
Mapping LGTM